config/apply/ccsm/translate.js

```diff
@@ -407,81 +399,88 @@ export function translateResponse(customApiResponse, patientData) {
     }
 
-    // Is findingType always provided?
-    let procedureText = procedureMappings[findingType?.ID];
-    let procedureCoding = standardProcedureCodes[procedureText];
+    // Skip add DiagnosticReport if there is no codings created
+    if (codings.length > 0) {
+      const conclusionCodes = [];
 
-    let conclusionCodes = [];
+      // Map the custom pap results to our standard codes
+      conclusionCodes = mapResults(papResults, customCytologyCodes, standardCytologyCodes, conclusionCodes);
 
-    // Map the custom pap results to our standard codes
-    conclusionCodes = mapResults(papResults, customCytologyCodes, standardCytologyCodes, conclusionCodes);
+      // Map the custom transformation zone results to our standard codes
+      if (transformationZone) {
+        const transformationZoneArray = [transformationZone];
+        conclusionCodes = mapResults(transformationZoneArray, customCytologyCodes, standardCytologyCodes, conclusionCodes);
+      }
+      // Map the custom HPV results to our standard codes
+      conclusionCodes = mapResults(hpvResults, customHpvCodes, standardHpvCodes, conclusionCodes);
 
-    // Map the custom transformation zone results to our standard codes
-    if (transformationZone) {
-      const transformationZoneArray = [transformationZone];
-      conclusionCodes = mapResults(transformationZoneArray, customCytologyCodes, standardCytologyCodes, conclusionCodes);
-    }
-    // Map the custom HPV results to our standard codes
-    conclusionCodes = mapResults(hpvResults, customHpvCodes, standardHpvCodes, conclusionCodes);
+      // Map the custom histology results to our standard codes
+      conclusionCodes = mapResults(colposcopyResults, customHistologyCodes, standardHistologyCodes, conclusionCodes);
+
+      // Map the custom ECC results to our standard codes
+      conclusionCodes = mapResults(endocervicalCuretageResults, customEccCodes, standardHistologyCodes, conclusionCodes);
+
+      // Map the custom excision results to our standard codes
+      if (ExcisionResultsShowAisOrCancer) {
+        conclusionCodes = mapResults(excisionResults, customExcisionCodes, standardHistologyCodes, conclusionCodes);
+      }
 
-    // Map the custom histology results to our standard codes
-    conclusionCodes = mapResults(colposcopyResults, customHistologyCodes, standardHistologyCodes, conclusionCodes);
+      // Create a DiagnosticReport resource from Order
+      const newDiagnosticReport = {
+        'resourceType': 'DiagnosticReport',
+        'id': orderId,
+        'status': 'final',
+        'subject': patient_reference,
+        'category': [
+          {
+            'coding': [
+              {
+                'system': 'http://terminology.hl7.org/CodeSystem/v2-0074',
+                'code': 'LAB'
+              }
+            ]
+          }
+        ],
+        'code': { 'coding': codings },
+        'conclusionCode': conclusionCodes,
+        'effectiveDateTime': order.ResultDate,
```

Comment:
Something else that just came to mind is that now that we are building our own DiagnosticReport as opposed to taking the DiagnosticReport from Epic, we may need to consider the cardinality of different fields from the customAPI. According to the [CustomAPI Mappings document](https://upmchs.sharepoint.com/sites/CDCMitreUPMCCollaboration/Shared%20Documents/Forms/AllItems.aspx?csf=1&web=1&e=ll2Mbv&cid=38a80f39%2D64a4%2D4abe%2D9b29%2D499f431068aa&FolderCTID=0x01200012484A92010F9A42BCD4C235B6BD15E4&id=%2Fsites%2FCDCMitreUPMCCollaboration%2FShared%20Documents%2FGeneral%2Ftech%5Fresources%2FUPMC%20Pap%20Tracking%20API%20Mapping%2FCustomAPI%20Mappings%2Emd&parent=%2Fsites%2FCDCMitreUPMCCollaboration%2FShared%20Documents%2FGeneral%2Ftech%5Fresources%2FUPMC%20Pap%20Tracking%20API%20Mapping), `ResultDate` is a field that could be null. It looks like regardless, the `order.ResultDate` field will exist, but it may just be null on occasion, making the dates in our DiagnosticReport and Procedures that we create `null`. We may want to consider adding handling for that scenario, maybe leaving out the date fields in the FHIR resources if there is no date sent by the customAPI.

I don't think this is very pressing since most customAPI responses we have seen do include dates, but it may be something we want to investigate in the future.

Reply:
Updated